net/nettest/pipe.go

```diff
@@ -38,17 +39,10 @@ func NewPipe(name string, maxBuf int) *Pipe {
 	return &Pipe{
 		name:   name,
 		maxBuf: maxBuf,
-		rCh:    make(chan struct{}, 1),
-		wCh:    make(chan struct{}, 1),
+		cnd:    sync.NewCond(&sync.Mutex{}),
```

Comment:
I wouldn't allocate a new mutex. I'd just add a `mu sync.Mutex` field to Pipe and here do:

```
p := &Pipe{..}
p.cnd = sync.NewCond(&p.mu)
```

then the `p.mu.Lock() + defer p.mu.Unlock()` look more normal